File: packages/shared-types/typescript/src/audit.ts
export interface AuditEntry {
  id: string;
  agent_id: string;
  task_id?: string | null;
  action: string;
  actor_user_id?: string | null;
  /** ISO 8601 timestamp */
  timestamp: string;
  correlation_id: string;
  parent_correlation_id?: string | null;
  details: Record<string, unknown>;
}

File: packages/shared-types/python/src/mycelium_shared_types/audit.py
"""AuditEntry — immutable log of an agent action."""

from __future__ import annotations

from datetime import datetime, timezone
from typing import Any, Optional

from pydantic import BaseModel, ConfigDict, Field


class AuditEntry(BaseModel):
    model_config = ConfigDict(extra="forbid")

    id: str
    agent_id: str
    task_id: Optional[str] = None
    action: str
    actor_user_id: Optional[str] = None
    timestamp: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))
    correlation_id: str
    parent_correlation_id: Optional[str] = None
    details: dict[str, Any] = Field(default_factory=dict)
